.cargo/config.toml

```diff
@@ -8,7 +8,7 @@ build-std = ["core", "compiler_builtins", "alloc"]
 # but don't change the README instructions as someone might not
 # have .cargo/
 [build]
-target = "svsm-target.json"
+target = "x86_64-unknown-none.json"
```

Comment:
Any reason for name change? Is it for build.rs?

Reply:
Yes, just to reinforce cclaudio's message, from a distro packaging POV it is highly desirable to avoid application specific build targets, if practical todo so. If we can successfully use x86_64-unknown-none, then gives us ability to have common toolchain packaging usable by multiple applications with similar needs to SVSM, instead of each having todo its own thing.

Reply:
Hmm, it's not just the name that makes it feasible for the distro though, but using the compiler built-in target. Then I can build `core` and `alloc` (with all their unstable/internal-only features) paired with the compiler. Someday cargo `-Zbuild-std` might become a stable option to build those for arbitrary targets, but not yet...

If you match the name while using your own spec -- well that might still find my build of `core`, but it won't have exactly the same target spec as your json. Are there significant differences you're holding out for?

Reply:
@cuviper The original `svsm-target.json` file defines the attribute `llvm-target: x86_64-unknown-none`, so I think that bindgen expects the file name to match the llvm-target. I would say that bindgen is generating bindings for the spec defined in the json file, but I don't know how to check that. This PR does not require any change to the spec (json file).

By the way, this is the error I get when I run bindgen without the patch applied.

 ```
   Compiling regex v1.7.1
   Compiling cexpr v0.6.0
   Compiling linux_svsm v0.1.0 (/tmp/linux-svsm)
error: failed to run custom build command for `linux_svsm v0.1.0 (/tmp/linux-svsm)`

Caused by:
  process didn't exit successfully: `/tmp/linux-svsm/target/debug/build/linux_svsm-62a4504e56acf8c9/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build.rs
  cargo:rerun-if-changed=external.h

  --- stderr
  thread 'main' panicked at 'libclang error; possible causes include:
  - Invalid flag syntax
  - Unrecognized flags
  - Invalid flag arguments
  - File I/O errors
  - Host vs. target architecture mismatch
  If you encounter an error missing from this list, please file an issue or a PR!', /home/cclaudio/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.63.0/ir/context.rs:530:15
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:111: target/svsm-target/debug/liblinux_svsm.a] Error 101
```

Reply:
The `llvm-target` is only loosely related, just enough that LLVM codegen knows what to do. It's not uncommon that a Rust target spec uses a "similar" `llvm-target` for that purpose. However, we still need Rust to be able to find its `core` crate especially, so I'm suggesting you move away from your json spec entirely and use the builtin Rust target.

Reply:
Bindgen picks up the name of the json file (i.e., `svsm-target`) as well as the target defined inside the json file and tries to use the json name instead of the actual underlying target. So, the generation of libclang would just fail as the target triplet (`svsm-target`) is not known to the libclang.

Reply:
In that case @cclaudio agreed, let's make things easier for bindgen and change the name

Reply:
I opened the issue #37 to document that